AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/DataStoreConfiguration.swift

```diff
@@ -0,0 +1,112 @@
+    public let local: Model
+    public let remote: Model
+}
+
+/// Conflict Handler function typealias. The function is used during a conflict that
+/// could not be resolved and requires a decision from the consumer.
+public typealias DataStoreConflictHandler = (DataStoreConclictData, DataStoreConflictHandlerResolver) -> Void
+
+/// Callback for the `DataStoreConflictHandler`.
+public typealias DataStoreConflictHandlerResolver = (DataStoreConflictHandlerResult) -> Void
+
+/// The conflict resolution result enum.
+public enum DataStoreConflictHandlerResult {
+
+    /// Discard the local changes in favor of the remote ones.
+    case discard
+
+    /// Keep the local changes.
+    case keep
+
+    /// Return a new `Model` instance that should used instead of the local and remote changes.
+    case new(Model)
+}
+
+/// The `DataStore` plugin configuration object.
+public struct DataStoreConfiguration {
+
+    /// A callback function called on unhandled errors
+    public let errorHandler: DataStoreErrorHandler
+    
+    /// A callback called when a conflict could not be resolved by the service
+    public let conflictHandler: DataStoreConflictHandler
+    
+    /// How often the sync engine will run (in seconds)
+    public let syncInterval: TimeInterval
+    
+    /// The number of records to sync per execution
+    public let syncMaxRecords: UInt
+    
+    /// The page size of each sync execution
+    public let syncPageSize: UInt
+
+    init(errorHandler: @escaping DataStoreErrorHandler,
+         conflictHandler: @escaping DataStoreConflictHandler,
+         syncInterval: TimeInterval,
+         syncMaxRecords: UInt,
+         syncPageSize: UInt) {
+        self.errorHandler = errorHandler
+        self.conflictHandler = conflictHandler
+        self.syncInterval = syncInterval
+        self.syncMaxRecords = syncMaxRecords
+        self.syncPageSize = syncPageSize
+    }
+
+}
+
+extension DataStoreConfiguration {
+
+    public static let defaultSyncInterval: TimeInterval = .hours(24)
+    public static let defaultSyncMaxRecords: UInt = 10_000
+    public static let defaultSyncPageSize: UInt = 1_000
+
+    /// Creates a custom configuration. The only required property is `conflictHandler`.
+    ///
+    /// - Parameters:
+    ///   - errorHandler: a callback function called on unhandled errors
+    ///   - conflictHandler: a callback called when a conflict could not be resolved by the service
+    ///   - syncInterval: how often the sync engine will run (in seconds)
+    ///   - syncMaxRecords: the number of records to sync per execution
+    ///   - syncPageSize: the page size of each sync execution
+    /// - Returns: an instance of `DataStoreConfiguration` with the passed parameters.
+    public static func custom(
+        errorHandler: @escaping DataStoreErrorHandler = { error in
+            Amplify.Logging.error(error: error)
+        },
+        conflictHandler: @escaping DataStoreConflictHandler,
```

Comment:
Seems like this line should be:
```
        conflictHandler: @escaping DataStoreConflictHandler = { _, resolve in
            resolve(.discard)
        },
```

In the event that a customer wants to take the default for both the errorHandler and conflictHandler, but wants other custom fields?
```
        let configuration = DataStoreConfiguration.custom(
           syncInterval: 1.0,
           syncMaxRecords: 5,
           syncPageSize: 10)
```

Reply:
done!

Reply:
lgtm!, resolving.